src/main/kotlin/org/rust/ide/intentions/CreateFunctionIntention.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapiext.isUnitTestMode
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.parentOfType
+import org.rust.cargo.project.workspace.PackageOrigin
+import org.rust.ide.presentation.renderInsertionSafe
+import org.rust.ide.utils.GenericConstraints
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.types.type
+import org.rust.openapiext.buildAndRunTemplate
+import org.rust.openapiext.createSmartPointer
+
+class CreateFunctionIntention : RsElementBaseIntentionAction<CreateFunctionIntention.Context>() {
+    override fun getFamilyName() = text
+
+    sealed class Context {
+        data class Function(val callExpr: RsCallExpr, val name: String, val module: RsMod) : Context()
+    }
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        val path = element.parentOfType<RsPath>()
+        val functionCall = path?.parentOfType<RsCallExpr>()
+        if (functionCall != null) {
```

Comment:
Let's also check that the path is an actual path of the call (to make it unavailable in a case like `foo(bar::baz/*caret*/))`.
This can be checked by `functionCall.expr.isAncestorOf(path)`. 
Let's also add a test for this case